libraries/lib-xml/XMLWriter.cpp

```diff
@@ -428,3 +432,183 @@ void XMLStringWriter::Write(const wxString &data)
+      Write(">");
+   }
+}
+
+void XMLUtf8BufferWriter::WriteAttr(const std::string_view& name, const Identifier& value) 
+{
+   const wxScopedCharBuffer utf8Value = value.GET().utf8_str();
+
+   WriteAttr(name, { utf8Value.data(), utf8Value.length() });
+}
+
+void XMLUtf8BufferWriter::WriteAttr(
+   const std::string_view& name, const std::string_view& value)
+{
+   assert(mInTag);
+
+   Write(" ");
+   Write(name);
+   Write("=\"");
+   WriteEscaped(value);
+   Write("\"");
+}
+
+void XMLUtf8BufferWriter::WriteAttr(const std::string_view& name, int value)
+{
+   WriteAttr(name, static_cast<long long>(value));
+}
+
+void XMLUtf8BufferWriter::WriteAttr(const std::string_view& name, bool value)
+{
+   WriteAttr(name, static_cast<long long>(value));
+}
+
+void XMLUtf8BufferWriter::WriteAttr(const std::string_view& name, long value)
+{
+   // long can be int or long long. Assume the longest!
+   WriteAttr(name, static_cast<long long>(value));
+}
+
+void XMLUtf8BufferWriter::WriteAttr(
+   const std::string_view& name, long long value)
+{
+   // -9223372036854775807 is the worst case
+   constexpr size_t bufferSize = 21;
+   char buffer[bufferSize];
+
+   const auto result = ToChars(buffer, buffer + bufferSize, value);
+
+   if (result.ec != std::errc())
+      THROW_INCONSISTENCY_EXCEPTION;
+
+   WriteAttr(name, std::string_view(buffer, result.ptr - buffer));
+}
+
+void XMLUtf8BufferWriter::WriteAttr(const std::string_view& name, size_t value)
+{
+   // Well, that maintains the original behavior
+   WriteAttr(name, static_cast<long long>(value));
+}
+
+void XMLUtf8BufferWriter::WriteAttr(
+   const std::string_view& name, float value, int digits /*= -1*/)
+{
+   constexpr size_t bufferSize = std::numeric_limits<float>::max_digits10 +
+                                 5 + // No constexpr log2 yet! example - e-308
+                                 3; // Dot, sing an 0 separator
+
+   char buffer[bufferSize];
+
+   const auto result = ToChars(buffer, buffer + bufferSize, value, digits);
+
+   if (result.ec != std::errc())
+      THROW_INCONSISTENCY_EXCEPTION;
+
+   WriteAttr(name, std::string_view(buffer, result.ptr - buffer));
+}
+
+void XMLUtf8BufferWriter::WriteAttr(
+   const std::string_view& name, double value, int digits /*= -1*/)
+{
+   constexpr size_t bufferSize = std::numeric_limits<double>::max_digits10 +
+                                 5 + // No constexpr log2 yet!
+                                 3;  // Dot, sing an 0 separator
+
+   char buffer[bufferSize];
+
+   const auto result = ToChars(buffer, buffer + bufferSize, value, digits);
+
+   if (result.ec != std::errc())
+      THROW_INCONSISTENCY_EXCEPTION;
+
+   WriteAttr(name, std::string_view(buffer, result.ptr - buffer));
+}
+
+void XMLUtf8BufferWriter::WriteData(const std::string_view& value)
+{
+   if (mInTag)
+   {
+      Write(">");
+      mInTag = false;
+   }
+
+   WriteEscaped(value);
+}
+
+void XMLUtf8BufferWriter::WriteSubTree(const std::string_view& value)
```

Comment:
I think this isn’t used

Reply:
I've just copied the "Writer" interface, but I agree that this can be dropped.